operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -1609,6 +1690,91 @@ func (rc *ReconciliationContext) CheckRollingRestart() result.ReconcileResult {
 	return result.Continue()
 }
 
+func (rc *ReconciliationContext) MaybeUpdateCondition(condition api.DatacenterCondition) bool {
```

Comment:
I think this Type and Status should be the args here, it would remove a couple lines from each call-site

Reply:
There could potentially be more fields on the `Condition` struct in the future. For example, a human readable string explaining why the condition was set. Some fields may be optional. Is it still worth putting the fields in the function signature?

Reply:
hmm maybe keep the single arg the same, then, but add a convenience constructor `new DatacenterCondition()` now, and `newDatacenterConditionWithReason()` later

what do you think?

Reply:
so `newDatacenterCondition()` takes the type and status as parameters? That works for me!

Reply:
oh I also suggested moving away from the word Update because we're not using the k8s client's Update function here